tools/syz-declextract/run.go

```diff
@@ -54,13 +64,43 @@ func main() {
 	}
 	close(files)
 
+	var allOut []string
+	rename := renameSyscall(*kernelDir + "/arch/") // some syscalls have different names and entry points and thus need to
+	// be renamed e.g. SYSCALL_DEFINE1(setuid16, old_uid_t, uid) is referred to in the .tbl file with setuid.
```

Comment:
Use of function closure make the code less explicit and non-standard. 
I would write it the more standard way, where we first read and build the replace table:
```
syscallNames := readSyscallNames(...)
``` 

and then replace based on this table:
```
     allOut = append(allOut, renameSyscall(line, syscallNames))
```

It should make the code easier to understand.

Reply:
Done!